app/services/audio.js

```diff
@@ -61,8 +59,17 @@ export default Service.extend({
 
   init() {
     this.get('hifi').on('audio-ended', () => this.finishedTrack());
+    Ember.$(window).on('beforeunload', () => {
+      if (this.get('currentAudio')) {
+        this.sendListenAction(this.get('currentAudio'), 'close');
+      }
+    });
     this._super(...arguments);
   },
+  
+  willDestroy() {
+    Ember.$(window).off('beforeunload');
```

Comment:
was thinking more about some other feature's teardown steps getting inadvertently wiped out by `$(window).off('beforeunload')` but if this only really matters during testing that's fine.